### ISSUE TITLE
Add inverse to "only abstract methods" filter

### ISSUE THREAD

**Issue by sebastianbergmann** [Roave/BetterReflection#408]
`ReflectionClass::getMethods()` accepts a bitmask for filtering. `IS_ABSTRACT` is supported but an `IS_CONCRETE` is missing.

**Comment by Ocramius** [comment]
@sebastianbergmann to clarify, is this suggesting a completely new constant, or is it something you've already seen somewhere?

**Comment by sebastianbergmann** [comment]
I am proposing a new constant, yes.

**Comment by roukmoute** [comment]
Isn't it just the opposite of `IS_ABSTRACT` ?

**Comment by asgrim** [comment]
I think (correct me if I'm wrong) but the intended use case is something like:

```php
$reflection->getMethods(\ReflectionMethod::IS_PUBLIC & ~\ReflectionMethod::IS_ABSTRACT);
```

or in plain English "find me all public methods that are not abstract".

This isn't possible at the moment because we simply take the integer (e.g. 256 for `IS_PUBLIC` and `&` it against the function's modifiers):

https://github.com/Roave/BetterReflection/blob/master/src/Reflection/ReflectionClass.php#L350-L357

Not sure if the best approach is to change the way the comparison is done (potentially breaking compat with core reflection?), make a new method to do this, or add a new constant...

**Comment by sebastianbergmann** [comment]
Yes, @asgrim is correct.

I apologize for suggesting a solution instead of describing the problem I would like to see solved.

**Comment by asgrim** [comment]
BTW, in the interim before a solution becomes available, something like this could work:

```php
$publicConcreteMethods = array_filter(
    $reflection->getMethods,
    function (ReflectionMethod $method) {
        return $method->isPublic() && !$method->isAbstract();
    }
);
```

I appreciate it's not as elegant. But makes me think, what about a callback solution anyway?

```php
$publicConcreteMethods = $reflection->filterMethods(function (ReflectionMethod $method) {
    return $method->isPublic() && !$method->isAbstract();
});
```

**Comment by funivan** [comment]
My 2 cents:
```php
$reflection->filterMethods(
 new MethodCompositeFilter(
  new MethodIsPublic(), new MethodNotAbstract()
 )
);
```
or even:
$reflection->filterMethods(new MethodIsPublic(), new MethodNotAbstract())
);

**Comment by Ocramius** [comment]
IMO an `array_filter()` very much sufficient here: closing here.
